index.js

```diff
@@ -15,21 +15,35 @@ async function run() {
             Name: core.getInput('ssm-path', { required: true }),
             Value: core.getInput('ssm-value', { required: true }),
             Type: core.getInput('ssm-value-type', { required: true }),
-            Overwrite: core.getInput('ssm-value-overwrite', { required: true }),
+            Overwrite: Boolean(core.getInput('ssm-value-overwrite', { required: true })),
             Description: core.getInput('ssm-value-description')
         }
         const keyId = core.getInput('ssm-kms-key-id')
+        const accessKey = core.getInput('aws-access-key')
         if (params['Type'] === "SecureString" && keyId !== '') {
             core.debug(`Setting the KeyId to ${keyId}`)
             params['KeyId'] = keyId
         }
-        var result = await ssm.putParameter(params)
-        core.debug(`Parameter details Version [${result.Version}] Tier [${result.Tier}]`)
+        core.debug('Checking AWS authentication')
+        if (accessKey !== '') {
+            core.debug('Updating AWS authentication')
+            AWS.config.update({
+                secretAccessKey: core.getInput('aws-secret-key'),
+                accessKeyId: core.getInput('aws-access-key'),
+                region: core.getInput('aws-region')
```

Comment:
This will probably be cleaner if you pull out the block and do something similar to the line 14-20.

Also accessKeyId and secretAccessKey are deprecated, i think you need to use the credentials block. Last i checked the AWS SDK documentation.

Reply:
Sorry for the silence here, I've been pretty busy at my day job. I think I can make this work with only a change to documentation by instructing to pass environment variables in your workflow. https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-environment.html

I'm going on vacation next week so I'll try and test this out after.